provider/cmd/pulumi-resource-aws-apigateway/provider.ts

```diff
@@ -43,6 +43,65 @@ async function constructRestAPI(name: string, inputs: pulumi.Inputs,
         urn: restAPI.urn,
         state: {
             url: restAPI.url,
+            api: restAPI.api.apply(api => ({
+                urn: api.urn,
+                id: api.id,
+                apiKeySource: api.apiKeySource,
+                arn: api.arn,
+                binaryMediaTypes: api.binaryMediaTypes,
+                body: api.body,
+                createdDate: api.createdDate,
+                description: api.description,
+                disableExecuteApiEndpoint: api.disableExecuteApiEndpoint,
+                endpointConfiguration: api.endpointConfiguration,
+                executionArn: api.executionArn,
+                minimumCompressionSize: api.minimumCompressionSize,
+                name: api.name,
+                parameters: api.parameters,
+                policy: api.policy,
+                rootResourceId: api.rootResourceId,
+                tags: api.tags,
+                tagsAll: api.tagsAll,
+            })),
+            deployment: restAPI.deployment.apply(deployment => ({
+                urn: deployment.urn,
+                id: deployment.id,
+                createdDate: deployment.createdDate,
+                description: deployment.description,
+                executionArn: deployment.executionArn,
+                invokeUrl: deployment.invokeUrl,
+                restApi: deployment.restApi,
+                stageDescription: deployment.stageDescription,
+                stageName: deployment.stageName,
+                triggers: deployment.triggers,
+                variables: deployment.variables,
+            })),
+            stage: restAPI.stage.apply(stage => ({
+                urn: stage.urn,
+                id: stage.id,
+                accessLogSettings: stage.accessLogSettings,
+                arn: stage.arn,
+                cacheClusterEnabled: stage.cacheClusterEnabled,
+                cacheClusterSize: stage.cacheClusterSize,
+                clientCertificateId: stage.clientCertificateId,
+                deployment: stage.deployment,
+                description: stage.description,
+                documentationVersion: stage.documentationVersion,
+                executionArn: stage.executionArn,
+                invokeUrl: stage.invokeUrl,
+                restApi: stage.restApi,
+                stageName: stage.stageName,
+                tags: stage.tags,
+                tagsAll: stage.tagsAll,
+                variables: stage.variables,
+                xrayTracingEnabled: stage.xrayTracingEnabled,
+            })),
+            apiPolicy: restAPI.apiPolicy?.apply(apiPolicy => ({
+                urn: apiPolicy?.urn,
+                id: apiPolicy?.id,
+                policy: apiPolicy?.policy,
+                restApiId: apiPolicy?.restApiId,
+            })),
```

Comment:
Is this unwrapping actually needed for this to work?  I expected these could just be `apiPolicy: restAPI.apiPolicy`. Cc @justinvp to confirm whether that should work correctly.

Reply:
The unwrapping shouldn't be necessary.

Reply:
As I understood, that did not work

Reply:
Yes - worked through a few different approaches here which didn't work:

```typescript
apiPolicy: restAPI.apiPolicy,
```

```typescript
apiPolicy: restAPI.apiPolicy.apply(policy => ({ ...api })),
```

Reply:
What about it didn't work? Was there an error you could share?